appservice/src/deploy/deployRunFromZip.ts

```diff
@@ -0,0 +1,165 @@
+    } else {
+        throw new Error(localize('NotAZipError', 'Path specified is not a folder or a zip file'));
+    }
+
+    try {
+        const blobService: azureStorage.BlobService = await createBlobService(client, storageAccount);
+        const blobUrl: string = await createBlobFromZip(blobService, zipFilePath);
+        outputChannel.appendLine(formatDeployLog(client, localize('deployStart', 'Starting deployment...')));
+        const WEBSITE_USE_ZIP: string = 'WEBSITE_USE_ZIP';
+        const appSettings: StringDictionary = await client.listApplicationSettings();
+        appSettings.properties[WEBSITE_USE_ZIP] = blobUrl;
+        await client.updateApplicationSettings(appSettings);
+    } catch (error) {
+        // tslint:disable-next-line:no-unsafe-any
+        if (error && error.response && error.response.body) {
+            // Autorest doesn't support plain/text as a MIME type, so we have to get the error message from the response body ourselves
+            // https://github.com/Azure/autorest/issues/1527
+            // tslint:disable-next-line:no-unsafe-any
+            throw new Error(error.response.body);
+        } else {
+            throw error;
+        }
+    } finally {
+        if (createdZip) {
+            await FileUtilities.deleteFile(zipFilePath);
+        }
+    }
+}
+
+function parseAzureResourceId(resourceId: string): { [key: string]: string } {
+    const invalidIdErr: Error = new Error('Invalid Account ID.');
+    const result: {} = {};
+
+    if (!resourceId || resourceId.length < 2 || resourceId.charAt(0) !== '/') {
+        throw invalidIdErr;
+    }
+
+    const parts: string[] = resourceId.substring(1).split('/');
+
+    if (parts.length % 2 !== 0) {
+        throw invalidIdErr;
+    }
+
+    for (let i: number = 0; i < parts.length; i += 2) {
+        const key: string = parts[i];
+        const value: string = parts[i + 1];
+
+        if (key === '' || value === '') {
+            throw invalidIdErr;
+        }
+
+        result[key] = value;
+    }
+
+    return result;
+}
+
+async function createBlobService(client: SiteClient, sa: StorageAccount): Promise<azureStorage.BlobService> {
+    const parsedId: { [key: string]: string } = parseAzureResourceId(sa.id);
+    const resourceGroups: string = 'resourceGroups';
+    const saResourceGroup: string = parsedId[resourceGroups];
+    const storageAccountKeys: StorageAccountListKeysResult = await client.listStorageAccountKeys(saResourceGroup, sa.name);
+    return azureStorage.createBlobService(sa.name, storageAccountKeys.keys[0].value);
+
+}
+
+async function createBlobFromZip(blobService: azureStorage.BlobService, zipFilePath: string): Promise<string> {
+    const containerName: string = 'azureappservice-run-from-zip';
+    const blobName: string = azureStorage.date.secondsFromNow(0).toLocaleString().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '').replace(/\s/g, ''); // get rid of any special characters and spaces in the timestamp
```

Comment:
It feels like the date will be illegible once you remove all the special characters and spaces. We should investigate the issue you ran into because blob names should support some special characters

Reply:
Don't use toLocateString() - the output will vary depending on the machine/user's locale, and you could end up with cases you don't expect.  I'd suggest toISOString() - 2011-10-05T14:48:00.000Z.

You could replace the colons with hyphens or something but otherwise the format will be consistent.